ios/bytesReplacingReader.go

```diff
@@ -143,7 +137,12 @@ func (r *singleSearchReplaceReplacer) GetSizingHints() (int, int, float64) {
 }
 
 func (r *singleSearchReplaceReplacer) Index(buf []byte) (int, []byte, []byte) {
-	return bytes.Index(buf, r.search), r.search, r.replace
+	switch {
+	case len(r.search) == 1:
```

Comment:
First thanks for experimenting, benchmarking and contributing! Really appreciate it!

On this particular topic, I have the following thoughts:
- golang system `bytes.Index` already has this `len == 1` optimization: https://golang.org/src/bytes/bytes.go
- golang system `bytes.Index` chooses RabinKarp algo (IIRC) over other algos, say, Boyer-Moore, for a balanced perf reason I suspect. While in some cases or scenarios BM might perform better than RK, but in general, RK is an all-rounder.
- That said, there is nothing preventing you from implementing a customized `BytesReplacer` to have whatever specific implementation you'd like to have to suit your needs - this was the original goal of the last patch, wasn't it?

Reply:
No problem! Happy to tinker with code like this.

I understand `bytes.Index` already has the optimizations in question. Thank you for noting this. 
After further testing, I have come to the conclusion that Rabin-Karp is indeed a better option for indexing here. 

I will be maintaining this fork in my own repository with use-case specific optimizations. If I find anything that improves the overall performance of the `ios` package, I'll be sure to make another PR and look to see what your thoughts are.


I'll be closing this PR. Thanks for all the valuable help and information!